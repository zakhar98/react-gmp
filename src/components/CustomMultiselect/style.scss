.multiselect {
  .MuiFormControl-root {
    width: 100%;

    .MuiInput-root.MuiInput-underline {
      &::before {
        display: none;
      }

      &::after {
        display: none;
      }
    }

    .MuiFormLabel-root.MuiInputLabel-root {
      color: #f65261;
      text-transform: uppercase;
      font-size: 18px;
      font-weight: 500;
      position: static;
      transition: none;
      transform: none;
    }

    .MuiSelect-icon {
      color: #f65261;
    }

    .MuiSelect-select.MuiSelect-selectMenu.MuiInputBase-input {
      border-radius: 4px;
      color: white;
      padding: 12px;
      background-color: rgba(255, 255, 255, .1);
      font-family: "Work Sans", sans-serif;
      font-weight: 400;
      font-size: 18px;
    }
  }

  &.error {
    .MuiSelect-select.MuiSelect-selectMenu.MuiInputBase-input {
      border: 2px solid #f65261;
    }
  }

  &__error-text {
    color: #f65261;
    font-size: 12px;
    padding-top: 4px;
  }
}

.MuiPaper-root.MuiMenu-paper.MuiPopover-paper {
  color: #ffffff;
  background-color: #232323;

  .MuiCheckbox-colorSecondary {
    color: #ffffff;
  }
}
