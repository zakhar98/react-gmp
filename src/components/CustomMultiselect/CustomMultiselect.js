import React from "react";
import PropTypes from "prop-types";
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';
import Checkbox from '@material-ui/core/Checkbox';
import "./style.scss";

export default function CustomMultiselect(
  {name, value, options, label, onChange, placeholder}) {

    const MenuProps = {
      variant: "menu",
      getContentAnchorEl: null
    };

    return (
      <div className="multiselect">
        <FormControl>
          <InputLabel id="demo-mutiple-checkbox-label">{label}</InputLabel>
          <Select
            labelId="demo-mutiple-checkbox-label"
            id="demo-mutiple-checkbox"
            multiple
            value={value}
            onChange={onChange}
            name={name}
            input={<Input />}
            MenuProps={MenuProps}
            renderValue={(selected) => {
              if (selected.length === 0) {
                return <em>Placeholder</em>;
              }
  
              return selected.join(', ');
            }}
          > 
            <MenuItem value="" disabled>
              {placeholder}
            </MenuItem>
            {options.map((option) => (
              <MenuItem key={option.value} value={option.value}>
                <Checkbox checked={value.indexOf(option.value) > -1} />
                <ListItemText primary={option.name} />
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </div>
    );
}

CustomMultiselect.propTypes = {
  name: PropTypes.string,
  value: PropTypes.array.isRequired,
  options: PropTypes.array.isRequired,
  label: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
};
