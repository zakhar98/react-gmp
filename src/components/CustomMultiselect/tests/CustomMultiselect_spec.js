import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CustomMultiselect from '../CustomMultiselect.js';
import * as formik from 'formik';

formik.useField = jest.fn(()=>[{}, {}, {}]);

describe('CustomMultiselect', () => {
  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer
      .render(
      <CustomMultiselect 
        label="test_label"
        fieldName="test field name"
        options={[]}
        placeholder="test placeholder"
      />);
    expect(tree).toMatchSnapshot();
  });
});
