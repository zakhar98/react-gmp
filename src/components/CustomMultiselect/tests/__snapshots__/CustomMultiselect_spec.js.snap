// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMultiselect renders correctly 1`] = `
<div
  className="multiselect "
>
  <WithStyles(ForwardRef(FormControl))>
    <WithStyles(ForwardRef(InputLabel))
      id="undefined-label"
    >
      test_label
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Select))
      MenuProps={
        Object {
          "getContentAnchorEl": null,
          "variant": "menu",
        }
      }
      input={<WithStyles(ForwardRef(Input)) />}
      labelId="undefined-label"
      multiple={true}
      onChange={[Function]}
      onClose={[Function]}
      renderValue={[Function]}
    >
      <WithStyles(ForwardRef(MenuItem))
        disabled={true}
        value=""
      >
        test placeholder
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </WithStyles(ForwardRef(FormControl))>
</div>
`;
